src/store/modules/TsModsModule.ts

```diff
@@ -192,6 +176,28 @@ export const TsModsModule = {
         },
         updateDeprecated(state, allMods: ThunderstoreMod[]) {
             state.deprecated = Deprecations.getDeprecatedPackageMap(allMods);
+        },
+        prewarmCacheMod(state: State, mods: ThunderstoreMod[]) {
+            const localState = new Map<string, CachedMod>(state.cache.entries());
```

Comment:
Just wondering if changing from mutating the cache Map to duplicating it might lead to performance issues on large profiles or race conditions if the commit is called in rapid succession 🤷🏻

Reply:
This change was to mitigate a performance issue because it directly affects the `cachedMod` getter - Whenever the cache updates, the getter does a refresh too.

It's a valid concern about race conditions. I think we have the following options:
1. Wrap in a lock
  - IMO less ideal, but makes it easy to manage
2. Diff the cache and only re-set the changes - I'm unsure if calling `.set` re-triggers an update if the value is the same.
3. Keep the behaviour the same as before (always just call `.set`)
4. Something else?